src/sentry/integrations/vercel/integration.py

```diff
@@ -150,6 +153,35 @@ def build_integration(self, state):
                 "installation_type": installation_type,
                 "webhook_id": webhook["id"],
             },
+            "post_install_data": {"user_id": state["user_id"]},
         }
 
         return integration
+
+    def post_install(self, integration, organization, extra=None):
+        # check if we have an installation already
+        if SentryAppInstallationForProvider.objects.filter(
+            organization=organization, provider="vercel"
+        ).exists():
```

Comment:
What would happen if the user created a vercel integration, and then deleted the internal sentry app we create for them? Would the user be able to reconnect vercel at that point?

Reply:
@markstory No, things would break. I was originally not going to handle this case but maybe it's worth tackling. There is a big warning not to delete the integration but who knows what people would do

Reply:
Actually,  think. it would be better to just delete this row when we delete the sentry app.

Reply:
@markstory I added a change in the destroyer to delete the `SentryAppInstallationForProvider` row when the `SentryAppInstallation`. We still need the warning, however, since the user might uninstall the `SentryApp` before uninstalling the first-party Vercel integration.